### PR TITLE
Node.js AWS Task #7 Solution

### PR DESCRIPTION
**Pull request by OHalahan**
**TASK 7.1 ✅**
**TASK 7.2 ✅**
**TASK 7.3 ✅**

## Evaluation criteria ✅

Provide your reviewers with the link to the repo, client application and URLs to execute the /import path of the import-service

**Integrated FE link**
https://d3ehhvbc9pzoid.cloudfront.net/admin/products
> PR for alerts: https://github.com/OHalahan/nodejs-aws-fe/pull/4/files
> PR for local storage: https://github.com/OHalahan/nodejs-aws-fe/pull/5/files

**Import lambda link:**
https://m4vi868ne8.execute-api.eu-west-1.amazonaws.com/dev/import?name=products.csv

1 - authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file ✅
3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the ✅ importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided. ✅
5 - update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage authorization_token = localStorage.getItem('authorization_token') ✅

> FE app sets correct authorization token by default: 
> **ohalahan:TEST_PASSWORD => Basic b2hhbGFoYW46VEVTVF9QQVNTV09SRA==**
> go to **Dev tools -> Application -> Local Storage** and change token to check **403** response
> image

## Additional (optional) tasks ✅
+1 - Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file ✅

**No Authorization header provided (401):**
image

**Incorrect credentials (403):**
image

### CODE REVIEW

File: authorization-service/handlers/basicAutorize.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  console.log('event: ', event);
+
+  if (event.type !== 'TOKEN') {
+    return Promise.reject('Unauthorized');
+  }
+
+  try {
+    const resource = event.methodArn;
+
+    const {authorizationToken} = event;
```
Comment (AndreyS55) [inline authorization-service/handlers/basicAutorize.ts:14]: I suggest you to throw 401 error if there is no `authorizationToken`

File: authorization-service/handlers/basicAutorize.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const [authType, encodedCreds] = authorizationToken.split(' ');
+
+    console.log('AUTH TYPE: ', authType);
+    console.log('ENCODED CREDENTIALS: ', encodedCreds);
+
+    let effect = 'Allow';
+    let username = '';
+    let password = '';
+
+    if (authType && encodedCreds) {
```
Comment (AndreyS55) [inline authorization-service/handlers/basicAutorize.ts:25]: I also suggest to check that `authType === Basic` and throw 401 error if it's not.

**Review by bhondu — APPROVED** [review]

File: authorization-service/handlers/basicAutorize.ts (hunk 1 above)
Comment (OHalahan) [reply authorization-service/handlers/basicAutorize.ts:14]: Thank you for noticing this.
In fact, I was doing this initially, but hometask states that 401 should be returned only when Authorization header is missing

incorrect tokens (e.g. not Basic or missing auth token) should return 403;
If I understood task correctly :|

File: authorization-service/handlers/basicAutorize.ts (hunk 2 above)
Comment (OHalahan) [reply authorization-service/handlers/basicAutorize.ts:25]: explained above.
thanks again

**Review by veranika-savanchuk — APPROVED** [review]

**Pull request merged**